py/misc.h

```diff
@@ -125,7 +125,18 @@ size_t m_get_peak_bytes_allocated(void);
 /** array helpers ***********************************************/
 
 // get the number of elements in a fixed-size array
+#ifdef __GNUC__
+// Under GNU C, the incorrect use of MP_ARRAY_SIZE on a pointer can be detected at build time
+#define _MP_IS_ARRAY(value) \
```

Comment:
Please don't prefix any names with `_` because that's reserved for the compiler.  Just `MP_IS_ARRAY` would be good, it does what the name says so could potentially be used elsewhere.